qiskit/providers/__init__.py

```diff
@@ -110,6 +109,381 @@
    Job
    JobV1
 
+======================
+Writing a New Provider
+======================
+
+If you have a Quantum device or simulator that you would like to integrate with
+Qiskit the way to go about this is writing a provider. A provider will provide
+Terra with a method to get available :class:`~qiskit.provider.BackendV1` objects
+The :class:`~qiskit.provider.BackendV1` object provides both information
+describing a backend and its operation for the :mod:`~qiskit.transpiler` so that
+circuits can be compiled to something that is optimized and can execute on the
+backend and also the :meth:`~qiskit.providers.BackenvV1.run` method which can
+run the :class:`~qiskit.circuit.QuantumCircuit` objects and/or
+:class:`~qiskit.pulse.Schedule` objects. This enables users and other Qiskit
+APIs, such as :func:`~qiskit.execute.execute` and higher level algorithms in
+Qiskit Aqua, to get results from executing circuits on devices in a standard
+fashion regardless of how the Backend is implemented. For a simple example of
+a provider the
+`qiskit-aqt-provider <https://github.com/qiskit-community/qiskit-aqt-provider>`__
+packge is worth using as an example.
+
+Provider
+========
+
+A provider class serves a single purpose to get backend objects that enable
```

Comment:
I am not sure if we want to say providers only do backends, even though that is all we got now.  For example, there could be a transpiler pass manager that gets served up.